service_worker/background.js

```diff
@@ -146,151 +104,114 @@ const hashPassword = async (password) => {
 
         const hash = await hashPassword(password);
 
-        // Check if the provided password matches the stored password
-        if (storedPassword === hash) {
+        if (storedPassword !== hash)
+            return sendResponse({ status: false, error: "Wrong password" });
+
+        try {
+            chrome.action.setIcon({
+                path: "../assets/Logo_inactive.png",
+            });
             removeServiceWorker();
             clearTimeout(timerId);
-            chrome.storage.local.set({loggedIn : false, sessionTimeout:false, timeLeft: 1 * 60 * 1000},)
+            clearInterval(intervalId)
+            await chrome.storage.local.set({ loggedIn: false, sessionTimeout: false })
 
-            console.log('User logged out successfully');
-            sendResponse({ success: true });
+            sendResponse({ status: true });
 
             await allowHttpsSearchAsync();
 
-            chrome.windows.getAll({ populate: true }, (windows) => {
-                windows.forEach((window) => {
-                    chrome.windows.remove(window.id);
-                });
+            const windows = await chrome.windows.getAll({ populate: true })
+            windows.forEach((window) => {
+                chrome.windows.remove(window.id);
             });
 
-            chrome.windows.create({
-                url: 'https://google.com',
-                type: 'normal'
-            });
-        } else {
-            sendResponse({ success: false, error: '*Invalid Credentials' });
+            await chrome.windows.create({ type: 'normal' });
+        } catch (error) {
+            sendResponse({ status: false, error: "Error logging out, try closing all the windows" })
         }
     } catch (error) {
-        console.error('Error logging out:', error);
-        sendResponse({ success: false, error: 'An error occurred while logging out' });
+        sendResponse({ status: false, error: "An error occurred while logging out" });
     }
 }
 
 export const injectServiceWorker = async (checkedToggles) => {
+
+    if (!checkedToggles) return;
     const rulesToInject = [];
     const oldRules = await chrome.declarativeNetRequest.getDynamicRules();
     const oldRulesIds = oldRules.map(rule => rule.id);
 
     rulesToInject.push(...defaultBlockRules);
 
     checkedToggles.forEach((toggle) => {
-        const {id}= toggle;
+        const { id } = toggle;
         switch (id) {
             case "socialMediaToggle":
-              rulesToInject.push(...socialMediaBlockRules);
-              break;
+                rulesToInject.push(...socialMediaBlockRules);
+                break;
             case "gamingToggle":
-              rulesToInject.push(...gamingSiteRules);
-              break;
+                rulesToInject.push(...gamingSiteRules);
+                break;
             default:
-              console.log("Unknown toggle status");
+                console.log("Unknown toggle status");
         }
```

Comment:
Move away all of the code which is not explicitly required here to a different function, for example, here you could move all the code to get old rules and rule id to a common func say... updateDynamicRules(rulesToInject : type -> list[]) because old rules and rule  ids are not needed to generate new rulesToInject list

So, you can generate this rulesToInject list here and move other things like updating dynamic rules and getting old rules there